(dsl): Support `Boosting` query (#364)
lambdaworks/zio-elasticsearch: docs/overview/queries/elastic_query_boost.md

---
id: elastic_query_boost
title: "Boosting Query"
---

> The `Boosting` query returns documents that match the positive query. Among those documents, the ones that also match the negative query get their relevance score lowered by multiplying it by the negative boosting factor. 

Reviewer: ```suggestion
The `Boosting` query returns documents that match the query marked as `positive` while reducing the relevance score of documents that also match a query which is marked as `negative` query. 
```
